x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations_on_remote.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, SavedObjectsClient, Logger } from '@kbn/core/server';
+
+import semverGte from 'semver/functions/gte';
+
+import type { PackageClient } from '../services';
+import { outputService } from '../services';
+
+import { PackageNotFoundError } from '../errors';
+
+import type { SyncIntegrationsData } from './sync_integrations_task';
+
+const FLEET_SYNCED_INTEGRATIONS_CCR_INDEX_PREFIX = 'fleet-synced-integrations-ccr-*';
+
+const MAX_RETRY_ATTEMPTS = 5;
+const RETRY_BACKOFF_MINUTES = [5, 10, 20, 40, 60];
+
+const getSyncedIntegrationsCCRDoc = async (
+  esClient: ElasticsearchClient,
+  abortController: AbortController
+): Promise<SyncIntegrationsData | undefined> => {
+  const indices = await esClient.indices.get(
+    {
+      index: FLEET_SYNCED_INTEGRATIONS_CCR_INDEX_PREFIX,
+    },
+    { signal: abortController.signal }
+  );
+
+  const indexNames = Object.keys(indices);
+  if (indexNames.length > 1) {
+    throw new Error(
+      `Not supported to sync multiple indices with prefix ${FLEET_SYNCED_INTEGRATIONS_CCR_INDEX_PREFIX}`
+    );
+  }
+
+  if (indexNames.length === 0) {
+    return undefined;
+  }
+
+  const response = await esClient.search(
+    {
+      index: indexNames[0],
+    },
+    { signal: abortController.signal }
+  );
+  if (response.hits.hits.length === 0) {
+    return undefined;
+  }
+  return response.hits.hits[0]._source as SyncIntegrationsData;
+};
+
+async function getSyncIntegrationsEnabled(
```

Comment:
I am wondering why this is needed and if this is not be user error prone, in my understanding if user use a remote elasticsearch (targer cluster B) on their cluster A, they will probably not configure an output in their cluster B if they are not using Fleet there no?

Reply:
This logic is there in case there are multiple remote clusters receiving the same sync integrations doc, I'm checking if the current remote cluster has sync enabled (matching the current elasticsearch host with the remote hosts coming from the management cluster)